### PR TITLE
test(core): Add unit tests for contains-products promotion condition

### PR DESCRIPTION
**Pull request by GabrielRoc**
## Description

Adds a direct unit test suite for the `contains_products` promotion condition (`contains-products-condition.ts`), which previously had **0% function coverage**.

The condition sums `line.quantity` for all order lines whose variant ID is in `productVariantIds`, then returns `args.minimum <= matches`.

6 tests covering:
- **Equivalence partitioning** over the match count: zero matches / below minimum / at minimum / above minimum
- Quantity accumulation across multiple eligible lines
- Correct exclusion of non-eligible lines

All tests are database-free.

## Breaking changes

None.

## Checklist

- [x] I have set a clear title
- [x] My PR is small and contains a single feature
- [x] I have added or updated test cases

Relates to #4835

---
<a href="https://app.blacksmith.sh/vendurehq/codesmith/vendure/pr/4881"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1785127839&installation_id=128408429&pr_number=4881&repository=vendurehq%2Fvendure&return_to=https%3A%2F%2Fgithub.com%2Fvendurehq%2Fvendure%2Fpull%2F4881&signature=ddfa92dc445aa04dbbff1f332fa55ea6909e669421052652f3605931e6755cc3"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

**Comment by GabrielRoc** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by GabrielRoc** [comment]
recheck

File: packages/core/src/config/promotion/conditions/contains-products-condition.spec.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import { ConfigArg } from '@vendure/common/lib/generated-types';
+import { ID } from '@vendure/common/lib/shared-types';
+import { describe, expect, it } from 'vitest';
+
+import { OrderLine } from '../../../entity/order-line/order-line.entity';
+import { Order } from '../../../entity/order/order.entity';
+import { ProductVariant } from '../../../entity/product-variant/product-variant.entity';
+import { createRequestContext } from '../../../testing/order-test-utils';
+
```
[the remark this answers is not in the document]
Comment (GabrielRoc) [reply packages/core/src/config/promotion/conditions/contains-products-condition.spec.ts:9]: Dismissed. Same reasoning as the parallel comment on the other PRs: `containsProducts.check()` does not use the `promotion` argument, and `undefined as any` is the intentional pattern shared across all promotion-condition specs in this series. No functional risk.

File: packages/core/src/config/promotion/conditions/contains-products-condition.spec.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+function orderWith(lines: OrderLine[]): Order {
+    return new Order({ lines });
+}
+
+const ctx = createRequestContext({ pricesIncludeTax: false });
+
+function check(order: Order, args: ConfigArg[]) {
+    return containsProducts.check(ctx, order, args, undefined as any);
+}
```
[the remark this answers is not in the document]
Comment (GabrielRoc) [reply packages/core/src/config/promotion/conditions/contains-products-condition.spec.ts:46]: Dismissed — duplicate of the comment above. The condition implementation does not touch the `promotion` argument; `undefined as any` is the intentional, consistent pattern used across all specs in this PR series.

File: packages/core/src/config/promotion/conditions/contains-products-condition.spec.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ * Unit tests for the `contains_products` promotion condition.
+ *
+ * The condition returns `true` when the total quantity of lines whose variant
+ * is in `productVariantIds` is at least `minimum`.
+ *
+ * Test cases are derived by:
+ *  - equivalence partitioning over the match count vs minimum
+ *  - boundary-value analysis at `matches = minimum ± 1`
+ *  - edge cases for multiple eligible lines and overlapping ID sets
+ */
```
[the remark this answers is not in the document]
Comment (GabrielRoc) [reply packages/core/src/config/promotion/conditions/contains-products-condition.spec.ts:22]: Valid point — the phrase was left over from an earlier draft. Fixed in commit `2f8690a46`: removed "and overlapping ID sets" from the JSDoc so the listed test derivation techniques accurately reflect the tests that exist.

**Review by grolmus — APPROVED** [review]
Excellent, well-structured coverage — boundary-value analysis on the thresholds, clean injector-based mocking (no DB), and assertions that the condition is invoked with the right args. Test-only, no production changes. Approving.
